addons/controllers/clusterbootstrap_controller.go

```diff
@@ -278,14 +319,271 @@ func (r *ClusterBootstrapReconciler) createOrPatchclusterBootstrapFromTemplate(c
 	return nil, errors.New("should not happen")
 }
 
-// createOrPatchPackageInstallSecret creates or patches or the secret used for PackageInstall in a cluster
-func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecret(cluster *clusterapiv1beta1.Cluster,
-	pkg *runtanzuv1alpha3.ClusterBootstrapPackage, clusterClient client.Client, namespace string, log logr.Logger) (*corev1.Secret, error) {
+// createOrPatchKappPackageInstall contains the logic that create/update PackageInstall CR for kapp-controller on
+// mgmt cluster. The kapp-controller running on mgmt cluster reconciles the PackageInstall CR and creates kapp-controller resources
+// on remote workload cluster. This is required for a workload cluster and its corresponding package installations to be functional.
+func (r *ClusterBootstrapReconciler) createOrPatchKappPackageInstall(clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap, cluster *clusterapiv1beta1.Cluster) error {
+	pkgi := &kapppkgiv1alpha1.PackageInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			// The legacy addons controller uses <cluster name>-<addon name> convention for naming the PackageInstall CR.
+			// https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/package_reconciler.go#L195.
+			// util.GeneratePackageInstallName() follows the same pattern.
+			Name: util.GeneratePackageInstallName(cluster.Name, util.GetPackageShortName(clusterBootstrap.Spec.Kapp.RefName)),
+			// kapp-controller PackageInstall CR is installed under the same namespace as tanzuClusterBootstrap. The namespace
+			// is also the same as where the cluster belongs.
+			Namespace: clusterBootstrap.Namespace,
+		},
+	}
+
+	pkgiMutateFn := func() error {
+		// TODO: Followup on the following fields, only add them if needed.
+		// https://github.com/vmware-tanzu/tanzu-framework/issues/1677
+		// if ipkg.ObjectMeta.Annotations == nil {
+		//	 ipkg.ObjectMeta.Annotations = make(map[string]string)
+		// }
+		// ipkg.ObjectMeta.Annotations[addontypes.YttMarkerAnnotation] = ""
+		// ipkg.Spec.SyncPeriod = &metav1.Duration{Duration: r.Config.AppSyncPeriod}
```

Comment:
Thought: We probably want to retain SyncPeriod to 5min similar to what was there in old addons controller.

Reply:
Yes, I left a TODO and filed an issue tracking this  https://github.com/vmware-tanzu/tanzu-framework/issues/1677. The reasons for leaving a TODO are because a) I guess it is not a must-to-have as MVP, I would follow up and add more fields like that if needed. b) after testing with real cluster, we would have better understanding on the user stories that requires the advanced configurations, and implement it all together.

Reply:
well, I think I have made some changes to support the sync period. Above TODO and issue will track whether we should preserve additional annotations.

Reply:
fixed